homeassistant/components/squeezebox/media_player.py

```diff
@@ -269,26 +251,9 @@ def state(self) -> MediaPlayerState | None:
         )
         return None
 
-    async def async_update(self) -> None:
-        """Update the Player() object."""
-        # only update available players, newly available players will be rediscovered and marked available
-        if self._attr_available:
-            last_media_position = self.media_position
-            await self._player.async_update()
-            if self.media_position != last_media_position:
-                self._last_update = utcnow()
-            if self._player.connected is False:
-                _LOGGER.debug("Player %s is not available", self.name)
-                self._attr_available = False
-
-                # start listening for restored players
-                self._remove_dispatcher = async_dispatcher_connect(
-                    self.hass, SIGNAL_PLAYER_REDISCOVERED, self.rediscovered
-                )
-
     async def async_will_remove_from_hass(self) -> None:
         """Remove from list of known players when removed from hass."""
-        self.hass.data[DOMAIN][KNOWN_PLAYERS].remove(self)
+        self.hass.data[DOMAIN][KNOWN_PLAYERS].remove(self.coordinator)
```

Comment:
I think it would be nice to have a think about how we use the known player list as _theoretically_, someone could disable their media player and then this code would be run and cause other side effects.

(And yes i know this number is practically 0, but i do think it's a good practice to avoid this in any case)

What do we use the known players for again?

Reply:
Every time we poll the server we get a list of connected players. We use known_players to decide if we already have coordinators and entities for all of these players or not. The relevant code is in __init__.py starting at line 138.

Reply:
Do we have something unique for those devices, could we store only the identifier? Also, follow up question, does the dispatcher stuff work when you have multiple squeezebox installations? Or will they both try to add the new discovered player

Reply:
A player can only be attached to one server at a time. Although I haven't tested this scenario locally, the logic should handle handing off the player from one server to another. Each time a new combination is detected it will be added to this list, and the combinations not being used will go inactive.

The reason I stored the Player object previously, and coordinator now, is that we want to keep track of the unique combination of Player and Server. We could store both of those uuids instead if you prefer.

In retrospect, known_players might be a slight misnaming, it is really known_player_connections or known_player_coordinators.

Reply:
> A player can only be attached to one server at a time

Yes, but we only use one dispatcher for all config entries right?

Reply:
> > A player can only be attached to one server at a time
> 
> Yes, but we only use one dispatcher for all config entries right?

That's right. I've pushed a new update. We now store only the uuid, not the object. I've added a known_servers key, under which we store known_players by server uuid. That way each pair is tracked but not by storing any objects.